packages/ckeditor5-image/src/imagecaption/imagecaptionediting.js

```diff
@@ -242,4 +246,36 @@ export default class ImageCaptionEditing extends Plugin {
 	_saveCaption( imageModelElement, caption ) {
 		this._savedCaptionsMap.set( imageModelElement, caption.toJSON() );
 	}
+
+	/**
+	 * Reconverts image caption when image alt attribute changes.
+	 * The change of alt attribute is reflected in caption's aria-label attribute.
+	 *
+	 * @private
+	 */
+	_registerCaptionReconversion() {
+		const model = this.editor.model;
+
+		model.document.on( 'change:data', () => {
```

Comment:
@niegowski Could you take a look at this bit please? For instance, I'm worried about about accessing `image.name` without checking whether it exists but I'm not an expert in reconversion.

Reply:
Accessing `change.range.start.nodeAfter` is safe there because the range of a Differ attribute change diff item always starts before the changed node. For an element attribute change it's: `[<el>]</el>` but for the text, the range is flat: `<$text>foo</$text>[<$text alt="x">bar</$text>]<$text>baz</$text>`. This makes `const caption = change.range.end.nodeAfter;` unsafe for cases where `alt` model attribute would be enabled (for whatever reason may be some 3rd party feature) on text nodes. So I would suggest to use a dedicated util to find the caption element: https://ckeditor.com/docs/ckeditor5/latest/api/module_image_imagecaptionutils_utils-ImageCaptionUtils.html#function-getCaptionFromImageModelElement.